python/ray/util/spark/cluster_init.py

```diff
@@ -793,9 +794,13 @@ def _verify_node_options(node_options, block_keys, node_type):
 @PublicAPI(stability="alpha")
 def setup_ray_cluster(
     num_worker_nodes: int,
+    *,
     num_cpus_per_node: Optional[int] = None,
+    num_cpus_head_node: Optional[int] = None,
```

Comment:
Do you think we can rename num_cpus_per_node to `num_cpus_worker_node` given it's still alpha?

Reply:
I changed argument name to `num_cpus_worker_node` and `num_gpus_worker_node`, our users are in POC stage so the change should be fine.

Reply:
for autoscaling, we can add a new argument "enable_autoscaling" (bool argument), if True then `num_worker_nodes` means the maximum number of worker nodes

Reply:
I added support for old argument `num_cpus_per_node` and `num_gpus_per_node` because few customers already use it in production.